libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_IP.c

```diff
@@ -2330,6 +2457,45 @@ BaseType_t FreeRTOS_IsNetworkUp( void )
 #endif
 /*-----------------------------------------------------------*/
 
+const char *FreeRTOS_strerror_r( BaseType_t xErrnum, char *pcBuffer, size_t uxLength )
+{
+const char *pcName;
+
+	switch( xErrnum )
+	{
+		case pdFREERTOS_ERRNO_EADDRINUSE:     pcName = "EADDRINUSE"; break;
+		case pdFREERTOS_ERRNO_ENOMEM:         pcName = "ENOMEM"; break;
+		case pdFREERTOS_ERRNO_EADDRNOTAVAIL:  pcName = "EADDRNOTAVAIL"; break;
+		case pdFREERTOS_ERRNO_ENOPROTOOPT:    pcName = "ENOPROTOOPT"; break;
+		case pdFREERTOS_ERRNO_EBADF:          pcName = "EBADF"; break;
+		case pdFREERTOS_ERRNO_ENOSPC:         pcName = "ENOSPC"; break;
+		case pdFREERTOS_ERRNO_ECANCELED:      pcName = "ECANCELED"; break;
+		case pdFREERTOS_ERRNO_ENOTCONN:       pcName = "ENOTCONN"; break;
+		case pdFREERTOS_ERRNO_EINPROGRESS:    pcName = "EINPROGRESS"; break;
+		case pdFREERTOS_ERRNO_EOPNOTSUPP:     pcName = "EOPNOTSUPP"; break;
+		case pdFREERTOS_ERRNO_EINTR:          pcName = "EINTR"; break;
+		case pdFREERTOS_ERRNO_ETIMEDOUT:      pcName = "ETIMEDOUT"; break;
+		case pdFREERTOS_ERRNO_EINVAL:         pcName = "EINVAL"; break;
+		case pdFREERTOS_ERRNO_EWOULDBLOCK:    pcName = "EWOULDBLOCK"; break; /* same as EAGAIN */
+		case pdFREERTOS_ERRNO_EISCONN:        pcName = "EISCONN"; break;
+		default:
+			( void ) snprintf( pcBuffer, uxLength, "Errno %ld", ( BaseType_t ) xErrnum );	/*lint !e586 function 'snprintf' is deprecated. [MISRA 2012 Rule 21.6, required]. */
```

Comment:
Why is `xErrnum` casted to `(BaseType_t)`? It is already a `BaseType_t` variable.